io/net/tcpdump.py

```diff
@@ -50,25 +52,66 @@ def setUp(self):
         smm = SoftwareManager()
         if not smm.check_installed("tcpdump") and not smm.install("tcpdump"):
             self.cancel("Can not install tcpdump")
+        self.obj = self.ping_check()
+        self.obj.start()
 
-    def test(self):
+    def ping_check(self):
         """
-        Performs the tcpdump test.
+        ping check to peer machine
         """
         cmd = "ping -I %s %s -c %s" % (self.iface, self.peer_ip, self.count)
-        obj = process.SubProcess(cmd, verbose=False, shell=True)
-        obj.start()
-        output_file = os.path.join(self.outputdir, 'tcpdump')
-        cmd = "tcpdump -i %s -n -c %s -w '%s'" % (self.iface, self.count,
-                                                  output_file)
-        for line in process.run(cmd, shell=True,
+        return process.SubProcess(cmd, verbose=False, shell=True)
+
+    def get_cmd_result(self, command):
+        """
+        get result of test
+        """
+        for line in process.run(command, shell=True,
                                 ignore_status=True).stderr.decode("utf-8") \
                                                    .splitlines():
-            if "packets dropped by interface" in line:
+            if "packets dropped by kernel" in line:
                 self.log.info(line)
                 if int(line[0]) >= (int(self.drop) * int(self.count) / 100):
                     self.fail("%s, more than %s percent" % (line, self.drop))
-        obj.stop()
+
+    def test(self):
+        """
+        Performs the tcpdump test.
+        """
+        output_file = os.path.join(self.outputdir, 'tcpdump')
+        cmd = "tcpdump -i %s -n -c %s -w '%s'" % (self.iface, self.count,
+                                                  output_file)
+        self.get_cmd_result(cmd)
+
+    def test_protocal(self):
```

Comment:
@smruti77 these can be handled via different yaml options, no need of different functions.

Also, -w option is needed for all options.